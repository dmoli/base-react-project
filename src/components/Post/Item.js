import React, { Component } from 'react';
import { FormattedRelative } from 'react-intl';
import styled from 'styled-components';
import { PropTypes } from 'prop-types';

class Item extends Component {
  render() {
    return (
      <div>
        <GameItem>
          <img
            width='305'
            height='350'
            className='juego-imagen'
            data-src='images/juegos/prueba-juego.jpg|images/juegos/prueba-juego.jpg'
          />
          <GameInfoContainer>
            <GameName>{this.props.title}</GameName>
            <GamePrice>$ {this.props.id}</GamePrice>
            <GamePriceAfter>$ {this.props.id}</GamePriceAfter>
            <p>
              <FormattedRelative
                value={Date.now()}
                updateInterval={36000}
              />
            </p>
            <GameWsContainer>
              <GameWsIcon className='icon-whatsapp' />
              <a href='tel:+55951098779'>+569 5109 8779</a>
            </GameWsContainer>
          </GameInfoContainer>
        </GameItem>
      </div>
    );
  }
}

Item.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

const GameItem = styled.article`
  margin: 0px auto 30px auto;
  display: flex;
  background: rgb(235, 238, 241);
`;

const GameInfoContainer = styled.section`
  flex: 1;
  padding-top: 18px;
  padding-left: 18px;
  padding-bottom: 18px;
`;

const GameInfoItem = styled.p`
  margin: 0px 0px 10px 0px;
  font-size: 1.5rem;
  color: #4c4c4c;
`;

const GameName = GameInfoItem.extend`
  font-size: 2rem;
`;

const GamePrice = styled.p`
  font-size: 1rem !important;
  text-decoration: line-through;
  margin: 0px 0px 3px 0px !important;
  color: #777 !important;
`;

const GamePriceAfter = styled.p`
  background: rgb(208, 64, 58);
  color: #fff;
  display: inline-block;
  padding: 4px 7px;
  border-radius: 4px;
`;

const GameWsContainer = styled.article`
  margin: 16px 0px 0px 0px !important;
`;

const GameWsIcon = styled.span`
  color: #2480ff !important;
  margin: 0px 3px 0px 0px;
`;

export default Item;
