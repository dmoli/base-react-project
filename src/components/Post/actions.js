import api from '../../api';

export const actionTypes = {
  SET_POST: 'SET_POST',
  SHOW_ERROR: 'SHOW_ERROR',
};

export const setPost = function setPost(posts) {
  return {
    type: actionTypes.SET_POST,
    payload: posts,
  };
};

const nextPageFailure = function nextPageFailure() {
  return {
    type: actionTypes.SHOW_ERROR,
  };
};

export const nextPage = function nextPage() {
  return async (dispatch, getState) => {
    try {
      const state = getState();
      const currentPage = state.posts.page;
      const posts = await api.posts.getList(currentPage);
      dispatch(
        setPost(posts),
      );
      return posts;
    } catch (e) {
      return dispatch(nextPageFailure());
    }
  };
};
