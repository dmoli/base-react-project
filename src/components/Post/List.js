import React, { Component } from 'react';

import Item from './Item';

class List extends Component {
  render() {
    let error = null;
    if (this.props.error) {
      error = <div className="error">Error</div>;
    }
    return (
      <section>
        {error}
        {this.props.posts
          .map(post => (
            <Item key={post.id} {...post} />
          ))
        }
      </section>
    );
  }
}

export default List;
