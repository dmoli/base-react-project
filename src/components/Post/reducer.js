import { combineReducers } from 'redux';

import { actionTypes } from './actions';
import initialState from '../../redux/state';

function pageReducer(state = initialState.posts.page, action = {}) {
  switch (action.type) {
    case actionTypes.SET_POST:
      return state + 1;
    default:
      return state;
  }
}

function entitiesReducer(state = initialState.posts.entities, action = {}) {
  switch (action.type) {
    case actionTypes.SET_POST:
      return state.concat(action.payload);
    default:
      return state;
  }
}

const postReducer = combineReducers({
  page: pageReducer,
  entities: entitiesReducer,
});

export default postReducer;
