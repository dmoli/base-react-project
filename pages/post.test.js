/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Post from './post';

describe('Page - Post', () => {
  it('rendered correctly', () => {
    const component = renderer.create(<Post locale='en' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
