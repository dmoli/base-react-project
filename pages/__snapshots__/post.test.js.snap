// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - Post rendered correctly 1`] = `
<div
  className="layout"
>
  <header
    className="sc-EHOje jjVuSd"
  >
    <figure
      className="test sc-bZQynM jVHFEX"
    >
      <img
        alt="Logo Primer Player"
        height="127"
        src="/static/logo.png"
        width="157"
      />
    </figure>
    <div>
      <span
        className="icon-menu sc-bdVaJa jFQTqy"
        onClick={[Function]}
      />
      <nav
        className="sc-bwzfXH gssRyn"
        open={false}
      >
        <ul
          className="sc-htpNat cuBxGo"
        >
          <li
            className="sc-bxivhb gLOCki"
          >
            <a
              className="sc-ifAKCX hOzByA"
              onClick={[Function]}
            >
              <span>
                Hi link home
              </span>
            </a>
          </li>
          <li
            className="sc-bxivhb gLOCki"
          >
            <a
              className="sc-ifAKCX hOzByA"
              onClick={[Function]}
            >
              <span>
                Hi link post
              </span>
            </a>
          </li>
          <li
            className="sc-bxivhb gLOCki"
          >
            <a
              className="sc-ifAKCX hOzByA"
              onClick={[Function]}
            >
              <span>
                Hi link rebass
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <div
    className="sc-gzVnrw iFBSaR"
  >
    <section
      className="sc-htoDjs kKjDSp"
    >
      <input
        className="buscador sc-dnqmqq jCixQC"
        placeholder="Busca tu juego..."
        type="text"
      />
      <span
        className="icon-search sc-iwsKbI eYhjXI"
      />
    </section>
    <h1>
      <span>
        Hi link home
      </span>
    </h1>
    <section />
    <button
      onClick={[Function]}
    >
      Pagination
    </button>
  </div>
  <footer
    className="sc-gZMcBi bwimHQ"
  >
    I\`m here to stay
  </footer>
</div>
`;
